<a href="https://colab.research.google.com/github/mohameds277/Machine_learning/blob/main/Machine_learning_and_data_science_assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Machine learning and data science ENCS5341 assignment 1

# Mohamed Shqierat 1190702


---




# **1-** Read the dataset and examine how many features and examples does it have?


In [14]:
#importing pandas library as pd alias
import pandas as pd


#extracting cars.csv data into pandas datafram
data_frame_cars = pd.read_csv('cars.csv')

#printing ALL csv data in frame forms
print(data_frame_cars)


      mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin  
0            70     USA  
1            70     USA  
2            70     USA  
3            70     USA  
4    

In [21]:
number_of_samples = len(data_frame_cars)
print(f' number of examples : {number_of_samples}')

number_of_features = data_frame_cars.shape[1]
print(f' number of features : {number_of_features}')

print("Features  ")
features = data_frame_cars.columns
for feature  in  features:
    print(feature)


 number of examples : 398
 number of features : 8
Features  
mpg
cylinders
displacement
horsepower
weight
acceleration
model_year
origin


**there are 398 example and 8 feature for each example  in cars.csv dataset file **







# **2-**Are there features with missing values? How many missing values are there in
# each one?




In [20]:
print(data_frame_cars.isnull().sum())

mpg             0
cylinders       0
displacement    0
horsepower      6
weight          0
acceleration    0
model_year      0
origin          2
dtype: int64


AS appears  above , there are 6  missing values  located in hourspower feature and 2 missing values in origin feature


---



# **3- Fill the missing values in each feature using a proper imputation method**